### PR TITLE
Add CI GitHub workflows

### PR DESCRIPTION
**Pull request by mslukebo**
This PR also bumps NerdBank.GitVersioning to 3.3.37, which is required to allow `dotnet build` to work on GitHub Actions.

### CODE REVIEW

**Comment by mslukebo** [comment]
Currently I set up one Action that runs the build and then tests. I created a single badge in the README that indicates if this Action completed succesfully (i.e. builds and all tests pass) for the develop branch.

Two questions we should have are:
1) Do we want badges for main, develop, or both?
2) Do we want two badges where one displays "Build Succeeds" (i.e. an Action that only runs dotnet build) and another that displays "Tests Pass" (i.e. an Action that both builds and runs), or just the combined one I created?

**Comment by mslukebo** [comment]
This PR now contains workflows for the following situations:
1) push to main -> build main
2) push to main -> build main and run tests
3) push to develop -> build develop
4) push to develop -> build develop and run tests
5) pull request into develop or main -> build PR branch and run tests

The last workflow is for CI (and is the one running for this PR). The first four are strictly so we can have different badges for all scenarios. The text inside a badge comes from the workflow's name, so we need separate workflows for main/develop.

Github actions do not support any form of nesting/templating of actions, so unfortunately the workflows are copy-pasted for the most part :(

**Review by trgibeau — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: 'PR CI'
+
+on:
+  pull_request:
+    branches: [ develop, main ]
```
Comment (zachnew-msft) [inline .github/workflows/ci.yml:5]: We don't need develop anymore

File: .github/workflows/develop_build_status.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+name: 'develop build status'
```
Comment (zachnew-msft) [inline .github/workflows/develop_build_status.yml:1]: We can remove develop specific collateral.

File: .github/workflows/develop_tests_status.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
+name: 'develop tests status'
```
Comment (zachnew-msft) [inline .github/workflows/develop_tests_status.yml:1]: See comment on develop status.

**Review by zachnew-msft — CHANGES_REQUESTED** [review]

File: .github/workflows/develop_tests_status.yml (hunk 3 above)
Comment (mslukebo) [reply .github/workflows/develop_tests_status.yml]: Removed this file

File: .github/workflows/develop_build_status.yml (hunk 2 above)
Comment (mslukebo) [reply .github/workflows/develop_build_status.yml]: Removed this file

File: .github/workflows/ci.yml (hunk 1 above)
Comment (mslukebo) [reply .github/workflows/ci.yml]: Removed

**Review by zachnew-msft — APPROVED** [review]

**Review by trgibeau — APPROVED** [review]

**Pull request merged**